hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -1190,8 +1177,8 @@ private void rollbackInflightCommits() {
     }
   }
 
-  private HoodieTable getTableAndInitCtx() {
-    // Create a Hoodie table which encapsulated the commits and files visible
+  private HoodieTable getTableAndInitCtx(JavaRDD<HoodieRecord<T>> records) {
+    startEmbeddedServerView(records.map(rec -> rec.getPartitionPath()).distinct().collect());
```

Comment:
what's the implication of this map and collect ? The records are not cached/persisted before this operation, will this action result in twice reading of data from the DFS ? If yes, there are 2 options a) Persist the records before this (although it will need to be shuffled anyways). b) Do this as part of a different action already in the DAG. WDYT ?

Reply:
see comment above